sqlglot/parser.py

```diff
@@ -1595,30 +1591,38 @@ def _parse_create_sequence(self, replace, properties) -> exp.Create | exp.Comman
                 seq.set("data_type", self._parse_types())
             elif self._match_text_seq("INCREMENT"):
                 self._match_text_seq("BY")
+                self._match_text_seq("=")
                 seq.set("increment", self._parse_term())
             elif self._match_text_seq("MINVALUE"):
                 seq.set("minvalue", self._parse_term())
             elif self._match_text_seq("MAXVALUE"):
                 seq.set("maxvalue", self._parse_term())
             elif self._match(TokenType.START_WITH) or self._match_text_seq("START"):
+                self._match_text_seq("=")
                 seq.set("start", self._parse_term())
             elif self._match_text_seq("CACHE"):
                 # T-SQL allows empty CACHE which is initialized dynamically
-                seq.set("cache", self._parse_term() or exp.Var(this=""))
+                cache: t.Optional[exp.Expression | bool] = self._parse_term()
+                if cache and not isinstance(cache, exp.Literal):
+                    self._retreat(self._index - 1)
+                    cache = True
+                seq.set("cache", cache)
             elif self._match_text_seq("OWNED", "BY"):
-                owned = None if self._match_text_seq("NONE") else self._parse_column()
+                # "OWNED BY NONE" is the default
+                owned = self._prev.text if self._match_text_seq("NONE") else self._parse_column()
```

Comment:
Imo there's no point in carrying around redundant info here, since `OWNED BY NONE` can be omitted. I'd personally just set it to `None`.